### PR TITLE
Keep fetching on empty pages

### PR DESCRIPTION
**Pull request by helfi92**
Resolves https://github.com/taskcluster/taskcluster-web/issues/67.

### CODE REVIEW

File: src/ConnectionLoader.js [hunk 1]
```diff
@@ -5,6 +5,24 @@ const FIRST = '$$FIRST$$';
 
 export default class ConnectionLoader {
   constructor(singleConnectionHandler) {
+    const fetch = async ({ connection, options, ...props }) => {
+      const result = connection
+        ? await singleConnectionHandler({ connection, options, ...props })
+        : await singleConnectionHandler({ options, ...props });
+
+      if (result.continuationToken && !result.items.length) {
```
Comment (djmitche) [inline src/ConnectionLoader.js]: The returned items are not always in a property named "items".  [Bug 1436212](https://bugzilla.mozilla.org/show_bug.cgi?id=1436212) has some background on why this is hard.

Comment (helfi92) [reply src/ConnectionLoader.js]: Every connection loader in `taskcluster-web-server` returns an items property (https://github.com/taskcluster/taskcluster-web-server/search?utf8=%E2%9C%93&q=items%3A&type=). I guess we can write tests to assert that each connection handler returns an object with an `items` fields.

**Comment by helfi92** [comment]
> To solve the first issue, I think the best fix is to pass the result property name (or maybe a function for extracting items from result) when building the ConnectionLoader.

I'm not sure I understand what you mean. The items are always under `result.items`. See [earlier comment](https://github.com/taskcluster/taskcluster-web-server/pull/46#discussion_r224484948).

Regarding the second issue, I agree that we shouldn't sweat over it right now. Maybe later if migration to postgres starts taking too long.

File: src/ConnectionLoader.js (hunk 1 above)
Comment (djmitche) [reply src/ConnectionLoader.js]: Oh, awesome.  I didn't know these were results from a loader.  One of the downsides of this heavily-functional style of programming is that it's not easy to follow control flow!

**Comment by djmitche** [comment]
OK, sounds good :)

**Pull request merged**